ssl-config-core/src/main/scala/com/typesafe/sslconfig/ssl/DefaultHostnameVerifier.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2015 - 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.typesafe.sslconfig.ssl
+
+import java.security.Principal
+import java.security.cert.{ Certificate, CertificateException, X509Certificate }
+
+import com.typesafe.sslconfig.util.LoggerFactory
+import javax.net.ssl.{ HostnameVerifier, SSLPeerUnverifiedException, SSLSession }
+import javax.security.auth.kerberos.KerberosPrincipal
+import sun.security.util.HostnameChecker
+
+@deprecated("DefaultHostnameVerifier has been deprecated and does nothing.  Please use the javax.net.debug system property.", "0.4.0")
+class DefaultHostnameVerifier(mkLogger: LoggerFactory) extends HostnameVerifier {
+  private val logger = mkLogger(getClass)
+
+  def hostnameChecker: HostnameChecker = {
+    logger.warn("DefaultHostnameVerifier has been deprecated and does nothing.  Please use the javax.net.debug system property.")
```

Comment:
what has this got to do with the javax.net.debug property?

Reply:
This functionality has been dropped in favour of out-the-box javax.net.debug functionality.

I kind of just spray-painted the whole API with deprecation warnings. I'm happy to change them to more accurate advice if you have any suggestions.

Reply:
This is related to https://github.com/playframework/playframework/commit/f35dd0946fb84c8eee623b0e3d3cdee3226118a6 and `play.ws.ssl.loose.acceptAnyCertificate` being disabled.